samples/scripts/package.json

```diff
@@ -0,0 +1,13 @@
+{
+  "name": "nodejs-docs-samples-iot-scripts",
```

Comment:
after taking a closer look - I am mostly sure these tests aren't getting run :) I would suggest:

- `npm install`-ing all the dependencies you need into `samples/package.json`
- I would up-level all the tests into a top level `samples/tests` directory.  `mocha` by default doesn't do a recursive search, and it's far easier to find all the tests in a top level `tests` dir

Reply:
Yikes, this is going to take me a while.

Reply:
as @JustinBeckwith says, the goal here is that running `npm test` within the `./samples` folder executes all tests.

the testing framework we use `mocha` is clever, and can handle nested folder just fine, you can do something like:

```
mocha samples/folder1/test samples/folder2/test
```

Reply:
What about execAsync?

Reply:
@gguuss for tests, I tend to lean towards using `spawnSync` or `execSync`, since performance isn't a major concern, and it eliminates the need for a closure. If you do need to run something async, `child_process.spawn` and `child_process.exec` are a great choice (Node.js API for subprocesses is pretty darn good).

Reply:
I'll try and refactor the tests that depend on async execution.